Update rocm6.2_internal_testing branch with changes from rocm6.1_internal_testing branch

shbiswas834 opened the pull request.

pnunna93 committed:
Skip pytest parameterize check for pytest > 7.4.2

Commit:
Enable ROCm signal iir utils

Commit:
Enable ROCm signal iir utils

Commit:
Revert "Enable ROCm signal iir utils"

This reverts commit cbeb5f5b74f682bcbba95f5b9f984dd8b63fc7c2.

Commit:
Fix scipy window compilation errors

Commit:
Update UT status

Commit:
Update UT status

Commit:
Fix flake8 issues

AdrianAbeyta committed:
Merge pull request #46 from ROCm/rocm6.1_SWDEV-446431

Fix scipy signal tests for ROCm (SWDEV-446431)

lcskrishna committed:
fix the parser for test summary

lcskrishna committed:
fix parser for test summary

AdrianAbeyta committed:
Merge pull request #47 from lcskrishna/cl/fix-swdev-446433

[SWDEV-446433] Fix the unit test parser in CuPy

lcskrishna committed:
fix the parser

AdrianAbeyta committed:
Merge pull request #48 from lcskrishna/cl/fix-swdev-446433

Fix the parser in run_tests_rocm.py

shbiswas834 committed:
Merge remote-tracking branch 'origin/rocm6.1_internal_testing' into merge-6.1-to-6.2

shbiswas834 committed:
Resolved merge conflicts from 6.1

shbiswas834 committed:
Hipsolver issue with hipsolverDnParams_t

lcskrishna commented:
Hi @shbiswas834 could you please check build failures in the CI.

lcskrishna commented on cupy_backends/hip/cupy_hipsolver.h:
```diff
@@ -17,9 +17,8 @@ hipsolverStatus_t cusolverGetProperty(hipLibraryPropertyType_t type, int* val) {
     return HIPSOLVER_STATUS_SUCCESS;
 }
 
-typedef enum hipsolverDnParams_t {};
```
@shbiswas834  @AdrianAbeyta  Is there a reason we deleted this logic? If you see this is not useful anymore in our latest releases, wrap it with some condition like this below. 
```
#if HIP_VERSION less than <version>
typedef enum hipsolverDnParams_t {};
#endif 
```

shbiswas834 replied:
I removed it since in the build log it showed the variable is declared in rocsolver and a redeclaration. I have updated this branch where I moved hipsolverDnParams_t under HIP_VERSION < 50631061
It passes all checks.

lcskrishna replied:
@shbiswas834 This will fail on versions between 5.0 - 6.1 with error saying not declared. 
Can we wrap this one seperately and don't move below 5.0. 
```
#if HIP_VERSION < <6.2 version number>
typedef enum hipsolverDnParams_t {};
#endif 

....
#if HIP_VERSION < 50631061
typedef hipsolverHandle_t hipsolverDnHandle_t;
typedef void* hipsolverGesvdjInfo_t;
typedef void* hipsolverSyevjInfo_t;
```

lcskrishna commented on cupy_backends/hip/cupy_hipsolver.h:
```diff
@@ -17,9 +17,8 @@ hipsolverStatus_t cusolverGetProperty(hipLibraryPropertyType_t type, int* val) {
-typedef enum hipsolverDnParams_t {};
 
-#if HIP_VERSION < 50631061
+#if HIP_VERSION < 60240094
```
Double check this one again, same as above.

shbiswas834 replied:
Addressed this above

lcskrishna reviewed (changes requested):
Few minor changes might be needed to take care of versioning stuff. Otherwise thanks for creating the PR, it looks good.

shbiswas834 committed:
update hipsolver with hip_version

shbiswas834 commented:
@lcskrishna @AdrianAbeyta  Please note the updated changes. I have updated this branch where I moved hipsolverDnParams_t under HIP_VERSION < 5063106. Verified the use of this variable from the rocm5.0 docker used for build tests.

shbiswas834 commented:
[Cupy unit test suit logs ](https://amdcloud-my.sharepoint.com/:t:/g/personal/shbiswas_amd_com/ERw94j2_6R5MtKrl4HYsa5IBcsTdPdGmYR-NEieLHn6ehQ?e=EOzjqw)

lcskrishna reviewed (changes requested):
Could you please update these final changes? Else everything looks good. After fixing the builds on all the versions, let's focus on unit-test cases separately.

shbiswas834 committed:
updated hip version for hipsolver

shbiswas834 commented:
Updates:

- ROCm 6.2 throws this error if hipsolverDnParams_t is declared
       ![image](https://github.com/ROCm/cupy/assets/160769007/1c1967f9-52c9-4c8e-8aa0-d953b194ae5f)
- hipsolverDnParams_t is required HIP_VERSION < 60240092 hence encased it under single condition
- HIP_VERSION acquired from docker container
- updated condition checked with dockers with ROCm 5.0, 6.1 & 6.2

lcskrishna committed:
support hipsolverDnParams_t for older versions.

lcskrishna commented:
@shbiswas834 we need to update conditions for only hipsolverDnParams_t. I have updated the code.

AdrianAbeyta reviewed (approved):
LGTM, all checks now passing. Thanks @shbiswas834 @lcskrishna
